base58/src/lib.rs

```diff
@@ -16,6 +16,15 @@
 #![allow(clippy::needless_question_mark)] // https://github.com/rust-bitcoin/rust-bitcoin/pull/2134
 #![allow(clippy::manual_range_contains)] // More readable than clippy's format.
 
+// We only support 32-bit and 64-bit targets.
+//
+// - We can't guarantee this library works on 16-bit targets.
+// - 128 bit machines don't exist yet but Rust does not implement `Into<u64>` for `usize`,
```

Comment:
A CHERI Rust effort defined their usize as 128-bits as they technically have 128-bit pointers.

https://github.com/rust-lang/rust/issues/65473

I don't believe there's a need to consider unsupported/broken platforms here, yet this comment could caveat better.

Considering the call is for usize to no longer directly align with pointer width for such cases, and considering the intent here, this check may be better implemented based off sizeof usize (which I believe can be done with the techniques of const assert)?

Sorry to throw in the irrelevant oddities. It's an interesting rabbit hole and it does kinda rear its head here. Feel free to close without comment.

Reply:
> Considering the call is for usize to no longer directly align with pointer width for such cases, and considering the intent here, this check may be better implemented based off sizeof usize (which I believe can be done with the techniques of const assert)?

Oh, this is a good point, we should most likely do that. We're not messing with pointer casts in any library, so this would provide CHERI support out of the box.